### PR TITLE
Add new CVar's for default weapons

### PR DESCRIPTION
**Pull request by wopox1337**
I doubt the need for these CVar's. Public opinion is needed
| CVar                               | Default | Min | Max          | Description                                    |
| :--------------------------------- | :-----: | :-: | :----------: | :--------------------------------------------- |
| mp_t_default_grenades              | ""        | "" | -           | The default grenades that the Ts will spawn with. |
| mp_t_give_player_knife             | 1         | 0  | 1           | Whether Terrorist player spawn with knife. |
| mp_t_default_weapons_primary       | ""        | "" | -           | The default primary (rifle) weapon that the Ts will spawn with. |
| mp_t_default_weapons_secondary     | "glock18" | "" | -           | The default secondary (pistol) weapon that the Ts will spawn with. |
| mp_ct_default_grenades             | ""        | "" | -           | The default grenades that the CTs will spawn with. |
| mp_ct_give_player_knife            | 1         | 0  | 1           | Whether Counter-Terrorist player spawn with knife. |
| mp_ct_default_weapons_primary      | ""        | "" | -           | The default primary (rifle) weapon that the CTs will spawn with. |
| mp_ct_default_weapons_secondary    | "usp"     | "" | -           | The default secondary (pistol) weapon that the CTs will spawn with. |

Also can use few weapons:
`mp_ct_default_weapons_secondary "weapon_usp deagle"`
`mp_ct_default_weapons_primary "ak47 m4a1 awp weapon_famas"`

TODO:
 * ~~Find the better way to reading CVar's~~

### CODE REVIEW

**Comment by Vaqtincha** [comment]
It's better to add the add_default_weapon <team> <weapon> command (stored in the array) instead of a hundred useless cvars.

**Comment by Vaqtincha** [comment]
Бред имхо.

File: regamedll/dlls/player.cpp [hunk 1]
```diff
@@ -1451,39 +1451,126 @@ void EXT_FUNC CBasePlayer::__API_HOOK(GiveDefaultItems)()
 	{
 	case CT:
 	{
-		if (!HasRestrictItem(ITEM_KNIFE, ITEM_TYPE_EQUIPPED)) {
-			GiveNamedItem("weapon_knife");
+#ifdef REGAMEDLL_ADD
+		const char s[2] = " ";
+		char* token;
+
+		token = strtok(ct_default_weapons_melee.string, s); // Getting melee list for CT
```
Comment (fl0werD) [inline regamedll/dlls/player.cpp]: Когда забываешь, что в кс 1 нож.

File: regamedll/dlls/player.cpp [hunk 2]
```diff
@@ -1458,24 +1458,126 @@ void EXT_FUNC CBasePlayer::__API_HOOK(GiveDefaultItems)()
...
+		token = strtok(ct_default_weapons_secondary.string, s); // Getting secondary list for CT
+		while (token != NULL) {
+			ItemID const iItemID = GetItemIdByName(token); // TODO: GetWeaponInfo
+
+			if (iItemID != ITEM_NONE && !HasRestrictItem(iItemID, ITEM_TYPE_EQUIPPED)
+				&& (iItemID == ITEM_P228 || iItemID == ITEM_DEAGLE || iItemID == ITEM_ELITE || iItemID == ITEM_FIVESEVEN
+					|| iItemID == ITEM_GLOCK || iItemID == ITEM_GLOCK18 || iItemID == ITEM_USP
+					)
+				) {
+				GiveWeapon(m_bIsVIP ? 12 : 24, token); // TODO: GetWeaponInfo
```
Comment (wopox1337) [inline regamedll/dlls/player.cpp]: it is also necessary to invent something with this

Comment (fl0werD) [reply regamedll/dlls/player.cpp]: get weaponid of the struct by https://github.com/s1lentq/ReGameDLL_CS/blob/49965644c3a23c1ce2afd6f3d5463fa9f6522d58/regamedll/dlls/weapontype.cpp#L521
and check by
https://github.com/s1lentq/ReGameDLL_CS/blob/49965644c3a23c1ce2afd6f3d5463fa9f6522d58/regamedll/dlls/weapontype.cpp#L444
or
https://github.com/s1lentq/ReGameDLL_CS/blob/49965644c3a23c1ce2afd6f3d5463fa9f6522d58/regamedll/dlls/weapontype.cpp#L476

File: regamedll/dlls/player.cpp (hunk 1 above)
Comment (wopox1337) [reply regamedll/dlls/player.cpp]: fixed :)

**Comment by wopox1337** [comment]
Done. Need review.

**Comment by Vaqtincha** [comment]
бесполезная трата времени. такое прекрасно делается через 3rd partly

**Pull request merged**